src/ThreeJsPanel.ts

```diff
@@ -308,6 +314,64 @@ export class ThreeJsPanel {
     this.axisCamera.position.set(-this.axisOffset[0], -this.axisOffset[1], this.axisScale * 2.0);
   }
 
+  orthoScreenPixelsToPhysicalUnits(pixels: number, physicalUnitsPerWorldUnit: number): number {
+    // At orthoScale = 0.5, the viewport is 1 world unit tall
+    const worldUnitsPerPixel = this.orthoScale * 2 / this.getHeight();
+    // Multiply by devicePixelRatio to convert from scaled CSS pixels to physical pixels
+    // (to account for high dpi monitors, e.g.). We didn't do this to height above because
+    // that value comes from three, which works in physical pixels.
+    return pixels * window.devicePixelRatio * worldUnitsPerPixel * physicalUnitsPerWorldUnit;
+  }
+
+  setupOrthoScaleBar(): void {
+    const orthoScaleBarStyle: Partial<CSSStyleDeclaration> = {
+      border: "1px solid white",
+      borderTop: "none",
+      height: "10px",
+      display: "none",
+      position: "absolute",
+      right: "20px",
+      bottom: "20px",
+      color: "white",
+      mixBlendMode: "difference",
+      textAlign: "right",
+      lineHeight: "0px",
+      fontFamily: "-apple-system, 'Segoe UI', 'Helvetica Neue', Helvetica, Arial, sans-serif",
+      boxSizing: "border-box",
+      paddingRight: "10px",
+    };
+    Object.assign(this.orthoScaleBarElement.style, orthoScaleBarStyle);
+    this.containerdiv.appendChild(this.orthoScaleBarElement);
+  }
+
+  updateOrthoScaleBar(scale: number, unit?: string): void {
+    // We want to find the largest round number of physical units that keeps the scale bar within this width on screen
+    const SCALE_BAR_MAX_WIDTH = 150;
+    // Convert max width to volume physical units
+    const physicalMaxWidth = this.orthoScreenPixelsToPhysicalUnits(SCALE_BAR_MAX_WIDTH, scale);
+    // Round off all but the most significant digit of worldSpaceMaxWidth
+    const digits = Math.floor(Math.log10(physicalMaxWidth));
+    const div10 = 10 ** digits;
+    const scaleValue = Math.floor(physicalMaxWidth / div10) * div10;
+    let scaleStr = scaleValue.toString();
```

Comment:
did you consider using `toFixed` instead of this log10 stuff?

Reply:
On second thought, `toFixed` might eliminate the need for the digits<1 check below but not the log10 math.

Reply:
Nevermind!